invisible_cities/reco/deconv_functions.py

```diff
@@ -210,15 +211,17 @@ def deconvolution_input(data        : Tuple[np.ndarray, ...],
                             weight      : np.ndarray
                            ) -> Tuple[np.ndarray, Tuple[np.ndarray, ...]]:
 
-        ranges = [[coord.min() - 1.5 * sw, coord.max() + 1.5 * sw] for coord, sw in zip(data, sample_width)]
-        if inter_method in (InterpolationMethod.linear, InterpolationMethod.cubic, InterpolationMethod.nearest):
-            allbins   = [np.arange(rang[0], rang[1] + np.finfo(np.float32).eps, sw) for rang, sw in zip(ranges, sample_width)]
-            Hs, edges = np.histogramdd(data, bins=allbins, normed=False, weights=weight)
-        elif inter_method is InterpolationMethod.nointerpolation:
+        eps    = np.finfo(np.float32).eps
+        ranges = [ [ coord.min() - 1.5 * sw
+                   , coord.max() + 1.5 * sw + eps]
+                   for coord, sw in zip(data, sample_width) ]
+        if inter_method is InterpolationMethod.nointerpolation:
             allbins   = [grid[in_range(grid, *rang)] for rang, grid in zip(ranges, det_grid)]
+            allbins   = [binedges_from_bincenters(bins) for bins in allbins]
             Hs, edges = np.histogramdd(data, bins=allbins, normed=False, weights=weight)
         else:
-            raise ValueError(f'inter_method {inter_method} is not a valid interpolatin mode.')
+            allbins   = [np.arange(*rang, sw) for rang, sw in zip(ranges, sample_width)]
+            Hs, edges = np.histogramdd(data, bins=allbins, normed=False, weights=weight)
```

Comment:
**Edit: Sorry I for some reason appeared to reference the previous commit so this comment may have been posted twice.**

Does this not imply that someone can pass anything in other than `InterpolationMethod.nointerpolation` and get the behaviour from the else statement?

Just in testing:
```
>>> inter_method = 10
>>> if inter_method is InterpolationMethod.nointerpolation:
...     print("No interpolation please :)")
... else:
...     print("interp it up")
...
interp it up
```

This seems like an 'optional' default that will not be intuitive, as it wont throw an error if the user has input an invalid interpolation method.

Reply:
Ah, I didn't see it when I replied to your comment, but in the outer function, we check that `inter_method` has the correct type, so it can only have one of the values of `InterpolationMethod`:
```
    if not isinstance(inter_method, InterpolationMethod):
        raise ValueError(f'inter_method {inter_method} is not a valid interpolation method.')
```

I will recover the corresponding test. I would also like to replace this explicit check with the `check_annotations` decorator, but it's not possible at the moment because of #900.